core/Pos/Binary/Core/Types.hs

```diff
@@ -61,3 +61,32 @@ instance Bi T.SharedSeed where
 instance Bi T.ChainDifficulty where
     get = label "ChainDifficulty" $ T.ChainDifficulty . getUnsignedVarInt <$> get
     put (T.ChainDifficulty c) = put (UnsignedVarInt c)
+
+instance Bi T.BlockVersionData where
+    get = label "BlockVersionData" $ do
```

Comment:
>  I don't really understand though why isn't it better to do

Because this way it's easy to confuse order of values. In version you can see in which order stuff is written and in which order it's read, which makes it easier to check that didn't make a mistake.

Also I think we shouldn't really care about errors here, because we have tests for serialization which should catch such errors easily.